[Kernel] Remove all syncs from STA & VSA kernels

Edenzzzz opened the pull request:
None of the `cudaStreamSynchronize` (blocks the CPU) or the `cudaDeviceSynchronize` (blocks all streams including comm) is useful, but they are very expensive. Precisions are identical before and after removing.

# STA
`python csrc/attn/benchmarks/bench_sta.py`
### Before PR
![(FastVideo)](https://github.com/user-attachments/assets/643e9ac9-9dee-4841-9f07-edf47e31403f)
### After PR
![iverage time for forward pass (ms) 195 96](https://github.com/user-attachments/assets/2059c854-931b-4998-8196-8639b62df085)

# VSA
`python csrc/attn/benchmarks/bench_vsa.py `
### Before PR
![image](https://github.com/user-attachments/assets/aadbe9bf-99b3-4b11-a4b7-85f2afc539b3)
### After PR
![Pasted Graphic 2](https://github.com/user-attachments/assets/5b8fa256-0e27-4c6a-8c2a-8a3e6937820d)

Edenzzzz committed:
fix

Edenzzzz committed:
remove sync

Edenzzzz commented:
cc @BrianChen1129 @jzhang38

Edenzzzz commented on csrc/attn/st_attn/st_attn_h100.cu:
```diff
@@ -5,8 +5,15 @@
 
-#define CLAMP(value, min, max) ((value) < (min) ? (min) : ((value) > (max) ? (max) : (value)))
-#define ABS(x) ((x) < 0 ? -(x) : (x))
+// #define CLAMP(value, min, max) ((value) < (min) ? (min) : ((value) > (max) ? (max) : (value)))
```
`#define` cannot trigger compiler optimizations.

BrianChen1129 commented:
Have you tried inference videos?

Edenzzzz commented:
> Have you tried inference videos for both STA and VSA?

The current inference scripts seem to be still from v1. I will try and I don't think it will be any different, as the precision tests pass.

BrianChen1129 commented:
> > Have you tried inference videos for both STA and VSA?
> 
> The current inference scripts seem to be still from v0. I will try and I don't think it will be any different, as the precision tests pass.

The script can be found [here](https://github.com/hao-ai-lab/FastVideo/blob/main/scripts/inference/v1_inference_wan_STA.sh). And I do think it should pass the e2e test to ensure no other issues.

Edenzzzz commented:
> Have you tried inference videos for both STA and VSA?

Do you have a precision test for VSA? e.g. against Flexattn

Edenzzzz committed:
minor fix expr type

jzhang38 commented:
Precision test for VSA: 

https://github.com/hao-ai-lab/FastVideo/blob/main/csrc/attn/test/test_block_sparse.py

Edenzzzz committed:
lower test mem requirements and rename

Edenzzzz commented:
@jzhang38 The precision for VSA is also identical.

Edenzzzz committed:
docs

Edenzzzz committed:
fix

Edenzzzz committed:
add device guard

SolitaryThinker commented:
could rebase with main and see if CI pass?

Edenzzzz committed:
trigger ci

SolitaryThinker commented:
We should keep vsa and sta enabled as there are other code that effects them. These tests also are very short.

Edenzzzz committed:
remove custom_image

Edenzzzz commented:
> We should keep vsa and sta enabled as there are other code that effects them. These tests also are very short.

Added them back and added precision tests. Note that `workflow_dispatch` can have at most 10 variables, so I removed `custom_image` and used hardcoded string. Should probably try a different approach later.

jzhang38 commented:
lgtm
